pkg/process/net/common.go

```diff
@@ -126,34 +129,64 @@ func (r *RemoteSysProbeUtil) GetProcStats(pids []int32) (*model.ProcStatsWithPer
 	return results, nil
 }
 
-// GetConnections returns a set of active network connections, retrieved from the system probe service
+// GetConnections returns a set of active network connections, retrieved from the system probe service, supports the
+// grpc server as well as the http server.
 func (r *RemoteSysProbeUtil) GetConnections(clientID string) (*model.Connections, error) {
-	req, err := http.NewRequest("GET", fmt.Sprintf("%s?client_id=%s", connectionsURL, clientID), nil)
-	if err != nil {
-		return nil, err
-	}
-
-	req.Header.Set("Accept", contentTypeProtobuf)
-	resp, err := r.httpClient.Do(req)
-	if err != nil {
-		return nil, err
-	}
-
-	defer resp.Body.Close()
-
-	if resp.StatusCode != http.StatusOK {
-		return nil, fmt.Errorf("conn request failed: Probe Path %s, url: %s, status code: %d", r.path, connectionsURL, resp.StatusCode)
-	}
-
-	body, err := io.ReadAll(resp.Body)
-	if err != nil {
-		return nil, err
-	}
-
-	contentType := resp.Header.Get("Content-type")
-	conns, err := netEncoding.GetUnmarshaler(contentType).Unmarshal(body)
-	if err != nil {
-		return nil, err
+	var conns *model.Connections
+	useGRPCServer := dd_config.SystemProbe.GetBool("service_monitoring_config.use_grpc")
```

Comment:
I think it will be better to populate the value from the configuration in the initialization of the network check, rather than accessing the configuration here